packages/contractkit/src/utils/tx-uri.test.ts

```diff
@@ -0,0 +1,52 @@
+import { Tx } from 'web3-core'
+import { CeloContract } from '../base'
+import { newKit } from '../kit'
+import { buildUri, parseUri } from './tx-uri'
+
+describe('URI utils', () => {
+  const kit = newKit('http://localhost:8545')
+  const recipient = '0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef'
+  const value = '100'
+  const simpleTransferTx: Tx = {
+    value,
+    to: recipient,
+  }
+  const simpleTransferUri = `celo:${recipient}?value=${value}`
+  let stableTokenTransferUri: string
+  let stableTokenTransferTx: Tx
+
+  beforeAll(async () => {
```

Comment:
could you add a test for a function without parameters? (or this will be only for transfers?)

Reply:
could a "-1" params, or a third param for a 2 params function be a problem? It's curiosity, no one will generate something that does not work

Reply:
any additional params outside of `0:abiParamLength` will end up in the tx object... see my question above